opencv_videoio fails to compile with Visual Studio 2008.

When compiling "modules/videoio/src/cap_mjpeg_decoder.cpp", it prompts error message:
"Error	101	error C2039: 'data' : is not a member of 'std::vector<_Ty>'	e:\CQT\Toolkits\opencv3.0\opencv\modules\videoio\src\cap_mjpeg_decoder.cpp	793".

In function  '''std::vector<char> MotionJpegCapture::readFrame(frame_iterator it)''' the following code does not compile:
'''m_file_stream.read(result.data(), chunk.m_size);'''

--- a/modules/videoio/src/cap_mjpeg_decoder.cpp
+++ b/modules/videoio/src/cap_mjpeg_decoder.cpp
@@ -790,7 +790,7 @@ std::vector<char> MotionJpegCapture::readFrame(frame_iterator it)
     result.reserve(chunk.m_size);
     result.resize(chunk.m_size);
 
-    m_file_stream.read(result.data(), chunk.m_size);
+    m_file_stream.read(&(result[0]), chunk.m_size); // result.data() failed with MSVS2008
 
     return result;
 }

